client/main/notification.js

```diff
@@ -0,0 +1,94 @@
+        this.model.on("change:index", this.movedown, this);
+        this.bottom = (10*(this.model.attributes.index+1)+50*this.model.attributes.index);
+        this.render();
+    },
+    render: function() {
+
+        this.$el.addClass("silky-notification");
+        
+        if (this.model.attributes.visible){
+            this.$el.css("visibility", "visible");        
+        }else{
+            this.$el.css("visibility", "hidden");
+        }
+        
+        this.secondtimer = null;
+        if (this.model.attributes.timeToFade > 0){
+            setTimeout(() => {
+                this.$el.find(".silky-notification-link").addClass("silky-notification-link-fadeout");
+                this.secondtimer = setTimeout(() => {
+                    this.model.set("visible" , false);
+                    this.model.set("timeToFade" , 0);
+                    this.model.parent.remove(this);
+                }, 2000);
+            }, this.model.attributes.timeToFade);
+        }
+        
+        var html = '';
+        
+        html += '<div class="silky-notification-box">';
+        html += '    <span class="silky-notification-content">';
+        html += '       <h1>';
+        html += this.model.title;
+        html += '       </h1>';
+        html +=  (this.model.attributes.content === null) ? '':this.model.attributes.content +'</br>';
+        html +=  (this.model.attributes.description === null) ? '':this.model.attributes.description +'</br>';
+        html +=  (this.model.attributes.errorMessage === null) ? '':this.model.attributes.errorMessage +'</br>';
+        html +=  (this.model.attributes.errorCause === null) ? '':this.model.attributes.errorCause;
+        html += '    </span>';
+        if (this.model.attributes.linkText !== ''){
+            html += '    <span class="silky-notification-link">';
+            html += this.model.attributes.linkText;
+            html += '    </span>';
+        }
+        html += '</div>';
+
+        this.$el.html(html);
+        
+        
+        
+        
+        var newbottom = (10*(this.model.attributes.index+1)+50*this.model.attributes.index);
+        var diff = Math.abs(this.bottom-newbottom);
+        var $box = this.$el.find(".silky-notification-box");
+        
+        if (this.bottom === newbottom){
+            $box.css("bottom", ""+newbottom+"px");
+        }else if (this.bottom > newbottom){
+            $box.css("bottom", ""+this.bottom+"px");
+            $box.animate({bottom: "-="+diff}, "slow");
+            this.bottom = newbottom;
+        }else if (this.bottom < newbottom){
+            $box.css("bottom", ""+this.bottom+"px");
+            $box.animate({bottom: "+="+diff}, "slow");
+            this.bottom = newbottom;
+        }
+    },
+    _linkClicked : function(event){
+        this.model.linkAction();
+        if (this.model.attributes.hideOnAction){
+            this.model.set("visible" , false);
+            this.$el.css("visibility", "hidden");
```

Comment:
so you don't want these lines here:

    this.$el.css("visibility", "hidden");
    this.render();

all you want to do is modify the model, and then let the `model.on('change', ... )` callback to handle the rendering. that way you keep all of the "updating the view from the model" code in the one place.

i think you've got this set up all correctly on line 16, so you can probably just remove these lines.

makes sense?